WebAPI/Controllers/ExchangeController.cs

```diff
@@ -16,24 +17,37 @@ namespace WebAPI.Controllers
     public class ExchangeController : BaseController
     {        
         private readonly IBlockChangesService _blockChangesService;
-        public ExchangeController(IBlockChangesService blockChangeService) : base()
+        private readonly INotificationService _notificationService;
+
+        public ExchangeController(IBlockChangesService blockChangeService, INotificationService notificationService) : base()
         {
-            _blockChangesService = blockChangeService;           
+            _blockChangesService = blockChangeService;
+            _notificationService = notificationService;
         }
 
         [HttpPost]
         [Route("exchangeConfirm")]
         public async Task<IActionResult> ExchangeConfirm([FromBody]ExchangeRequestModel request)
         {
-            var blockChangeRequest = new BlockChangeRequest();
-            blockChangeRequest.BlockFrom = BlockForExchangeModel.ConvertToBlock(request.BlockFrom);
-            blockChangeRequest.BlockTo = BlockForExchangeModel.ConvertToBlock(request.BlockTo);
-            blockChangeRequest.Status = ExchangeStatus.WaitingForExchange;
-            blockChangeRequest.DateOfCreation = DateTime.Now;
-            blockChangeRequest.StudentId = Guid.Parse(request.StudentId);
+            var currentUserBlockChangeRequest = new BlockChangeRequest();
+            currentUserBlockChangeRequest.BlockFrom = BlockForExchangeModel.ConvertToBlock(request.BlockFrom);
+            currentUserBlockChangeRequest.BlockTo = BlockForExchangeModel.ConvertToBlock(request.BlockTo);
+            currentUserBlockChangeRequest.Status = ExchangeStatus.WaitingForExchange;
+            currentUserBlockChangeRequest.DateOfCreation = DateTime.Now;
+            currentUserBlockChangeRequest.StudentId = Guid.Parse(request.StudentId);
+
+            var (exchangeWasMade, foundMatch) = await _blockChangesService.AddAndFindMatch(currentUserBlockChangeRequest);
 
-            var res = await _blockChangesService.AddAndFindMatch(blockChangeRequest);
-            return Ok(res);
+            if (exchangeWasMade)
```

Comment:
Controller should be responsible for generating notification? May be it should be the task for changeservice